### PR TITLE
Free memory used for last stripe read when re-scanning a columnar table

### PR DESCRIPTION
**Pull request by onurctirtir**
(Rebased onto #5058 since using a function from there)

Instead of setting stripeReadState to NULL, call ColumnarResetRead
before re-scanning a columnar table since this function is already
designed for doing the necessary clean up when finishing a stripe
read.

Note that this change shouldn't have a great effect on memory usage
since AdvanceStripe was already doing the clean-up for all the
stripes except the last one.

DESCRIPTION: Reduces memory usage of columnar table scans by freeing the memory used for last stripe read

### CODE REVIEW

**Review by thanodnl — APPROVED** [review]
great catch.

I believe when a query uses rescanning, it might actually do it dozens of times, causing many last read states to leak. Cleaning them up will definitely help with keeping memory usage under control.

**Comment by onurctirtir** [comment]
Thank you @thanodnl for reviewing.
I will not merge this pr until merging #5058, since this pr is rebased onto that, fyi.

**Review by jeff-davis — APPROVED** [review]

**Pull request merged**
